meta-client/src/main/java/org/apache/helix/metaclient/recipes/lock/LockInfo.java

```diff
@@ -0,0 +1,193 @@
+    _record = new ZNRecord(ZNODE_ID);
+    setLockInfoFields(DEFAULT_LOCK_ID_TEXT, DEFAULT_OWNER_ID_TEXT, DEFAULT_CLIENT_ID_TEXT, DEFAULT_CLIENT_DATA, DEFAULT_GRANTED_AT_LONG,
+        DEFAULT_LAST_RENEWED_AT_LONG, DEFAULT_TIMEOUT_DURATION);
+  }
+
+  /**
+   * Initialize a LockInfo with a ZNRecord, set all info fields to default data
+   * @param znRecord The ZNRecord contains lock node data that used to initialize the LockInfo
+   */
+  public LockInfo(ZNRecord znRecord) {
+    this();
+    if (znRecord != null) {
+      String lockId = znRecord.getSimpleField(LockInfoAttribute.LOCK_ID.name());
+      String ownerId = znRecord.getSimpleField(LockInfoAttribute.OWNER_ID.name());
+      String clientId = znRecord.getSimpleField(LockInfoAttribute.CLIENT_ID.name());
+      String clientData = znRecord.getSimpleField(LockInfoAttribute.CLIENT_DATA.name());
+      long grantTime = znRecord.getLongField(LockInfoAttribute.GRANTED_AT.name(), DEFAULT_GRANTED_AT_LONG);
+      long lastRenewalTime =
+          znRecord.getLongField(LockInfoAttribute.LAST_RENEWED_AT.name(), DEFAULT_LAST_RENEWED_AT_LONG);
+      long timeout = znRecord.getLongField(LockInfoAttribute.TIMEOUT.name(), DEFAULT_TIMEOUT_DURATION.toMillis());
+      Duration duration = Duration.of(timeout, ChronoUnit.MILLIS);
+      setLockInfoFields(lockId,ownerId, clientId, clientData, grantTime,
+          lastRenewalTime, duration);
+    }
+  }
+
+  /**
+   * Initialize a LockInfo with data for each field, set all null info fields to default data
+   * @param lockId value of LOCK_ID attribute
+   * @param ownerId value of OWNER_ID attribute
+   * @param clientId value of CLIENT_ID attribute
+   * @param clientData value of CLIENT_DATA attribute
+   * @param grantTime the time the lock was granted
+   * @param lastRenewalTime the last time the lock was renewed
+   * @param timeout value of TIMEOUT attribute
+   */
+  public LockInfo(String lockId, String ownerId, String clientId,
+                  String clientData, long grantTime, long lastRenewalTime, Duration timeout) {
+    this();
+    setLockInfoFields(lockId, ownerId, clientId, clientData, grantTime, lastRenewalTime, timeout);
+  }
+
+  /**
+   * Set each field of lock info to user provided values if the values
+   * are not null, null values are set to default values
+   * @param lockId value of LOCK_ID attribute
+   * @param ownerId value of OWNER_ID attribute
+   * @param clientId value of CLIENT_ID attribute
+   * @param clientData value of CLIENT_DATA attribute
+   * @param grantTime the time the lock was granted
+   * @param lastRenewalTime the last time the lock was renewed
+   * @param timeout value of TIMEOUT attribute
+   */
+  private void setLockInfoFields(String lockId, String ownerId, String clientId, String clientData, long grantTime, long lastRenewalTime,
+                                 Duration timeout) {
+    _record.setSimpleField(LockInfoAttribute.LOCK_ID.name(), lockId == null ? DEFAULT_LOCK_ID_TEXT : lockId);
```

Comment:
Let's convert these lines into individual setters of each field, I think we may need them in lock client internal logic. Our goal is to completely wrap the ZNRecord so the external classes do not need to know that it is internally implemented using a ZNRecord.

Reply:
I am not sure I understand what you mean, do you mean having a separate method to set each of these values?

Reply:
Yes, I meant that we can have a setter method for each of the values, right now you already have getters.

Reply:
Got it, thanks!